external/Drivers/Sensor/Bosch-Sensortec/Driver_Sensor.c

```diff
@@ -0,0 +1,824 @@
+
+    bma2x2_read_accel_xyz(&acc);    //get acceleration data from sensor
+
+    pxyzData->X = acc.x;
+    pxyzData->Y = acc.y;
+    pxyzData->Z = acc.z;
+
+    //add timestamp here!
+    pxyzData->timeStamp = RTC_GetCounter();
+
+    return ARM_DRIVER_OK;
+}
+
+/****************************************************************************************************
+ * @fn      Accx_Deactivate
+ *          Deactivate Accelerometer
+ *
+ * @param   priv    Private Argument (Unused)
+ *
+ * @return  Return code: On success return ARM_DRIVER_OK
+ *
+ ***************************************************************************************************/
+static int32_t Accx_Deactivate (void *priv)
+{
+    bma2x2_set_Int_Enable( BMA2x2_DATA_EN, 0 );
+    NVIC_DisableIRQ(ACCEL_PINT_IRQn);
+    return ARM_DRIVER_OK;
+}
+
+/****************************************************************************************************
+ * @fn      Accx_PowerControl
+ *          Control the power state of the sensor
+ *
+ * @param   state   Power state
+ * @param   priv    Private Argument (Unused)
+ *
+ * @return  Return code: On success return ARM_DRIVER_OK
+ *
+ ***************************************************************************************************/
+static int32_t Accx_PowerControl(ARM_POWER_STATE state, void *priv)
+{
+    switch (state)
+    {
+        case ARM_POWER_OFF:
+            Accx_Deactivate(NULL);
+            break;
+
+        case ARM_POWER_LOW:
+            bma2x2_set_mode(BMA2x2_MODE_STANDBY);
+            break;
+
+        case ARM_POWER_FULL:
+            Accx_Activate(true,0,0,0,NULL);
+            break;
+
+        default:
+            return ARM_DRIVER_ERROR_PARAMETER;
+    }
+    return ARM_DRIVER_OK;
+}
+
+/****************************************************************************************************
+ * @fn      Accx_GetDetails
+ *          Get sensor details
+ *
+ * @param   data    Pointer to the sensor details
+ * @param   priv    Private Argument (Unused)
+ *
+ * @return  Returns ARM_DRIVER_ERROR_UNSUPPORTED since function is currently not supported
+ *
+ ***************************************************************************************************/
+static int32_t Accx_GetDetails (void *data, void *priv)
+{
+    return ARM_DRIVER_ERROR_UNSUPPORTED;
+}
+
+/****************************************************************************************************
+ * @fn      Accx_InjectData
+ *          Feed dummy data
+ *
+ * @param   data    Simulated data
+ * @param   priv    Private Argument (Unused)
+ *
+ * @return  Returns ARM_DRIVER_ERROR_UNSUPPORTED since function is currently not supported
+ *
+ ***************************************************************************************************/
+static int32_t Accx_InjectData (void *data, void *priv)
+{
+    return ARM_DRIVER_ERROR_UNSUPPORTED;    //Do nothing
+}
+
+/* Accel Driver Definition */
+OSP_BUILD_DRIVER_SENSOR(Acc, Accx, NULL);
```

Comment:
As discussed with you, I'm moving the APIs of Accel, Mag and Gyro (alongwith build macro definition) in three different files.
We discussed about the prefixes internally and have now declarer the build macro as `OSP_BUILD_DRIVER_SENSOR( Accel, OSP_Accel, NULL )`